### ISSUE TITLE
Add type to plugin tasks

### ISSUE THREAD

**Issue by JavierSegoviaCordoba** [gradle-nexus/publish-plugin#216]
`publishingProject.tasks.register(...) { }`

As it is not using any type, I can't filter with `withType` if I want to disable the publishing tasks, and by name, it can have tons of tasks.

**Comment by szpak** [comment]
Yes, it was just an umbrella tasks to simplify task orchestration. Would you see it as a new custom type or rather reuse something existing from Gradle?

**Comment by JavierSegoviaCordoba** [comment]
I am unsure if it is possible to reuse `PublishToMavenRepository` maybe a combination of both, `PublishToSonatype` extending `PublishToMavenRepository`, so it can be configured in a more granular way by the consumer.

Adding an option to the plugin extension, such as `disableIf { ... }`, can also be great.

Another workaround I am using is changing the root project version in order to allow publishing specific subprojects based on tag prefixes with different versions, but this would be a different issue (I think there is one already opened).
